### PR TITLE
feat(features/cards): card with grid and icon-flag

### PR DESCRIPTION
**Pull request by meowsun**
Card-item with grid and icon. There is no popup here. Need to check up and add popup later.

### CODE REVIEW

File: src/features/cards/organisms/card-item.js [hunk 1]
```diff
@@ -29,29 +83,115 @@ CardItem.propTypes = {
...
+)
+
+CardFlagWithNumber.propTypes = {
+  onUsefulClick: PropTypes.func.isRequired,
+  usefulFor: PropTypes.number.isRequired,
+  isUseful: PropTypes.bool.isRequired,
+}
+
+const FlagToggler = ({ isUseful, onUsefulClick }) => {
+  // const [isUseful, setIsUseful] = useState(usefulFor)
```
Comment (sergeysova) [inline src/features/cards/organisms/card-item.js]: Please, remove old code in comments

File: src/features/cards/organisms/card-item.js [hunk 2]
```diff
@@ -29,29 +83,115 @@ CardItem.propTypes = {
...
+
+  return (
+    <IconWrapper
+      onClick={() => {
```
Comment (sergeysova) [inline src/features/cards/organisms/card-item.js]:
```jsx
<IconWrapper onClick={onUsefulClick}>
```

File: src/features/cards/organisms/card-item.js [hunk 3]
```diff
@@ -29,29 +83,115 @@ CardItem.propTypes = {
...
-      ) : (
-        <Button small title="Add to saved" onClick={onUsefulClick}>
-          Save
-        </Button>
-      )}
-      {card.usefulFor > 0 ? <div>{card.usefulFor}</div> : <div>&nbsp;</div>}
+    <Row basis="25%" justify="flex-end" gap="1.4em" align="center">
       {card.meta.canEdit && <Link to={`/edit/${card.id}`}>Edit</Link>}
-      <i>{format(new Date(card.createdAt), "HH:MM MM/DD/YYYY")}</i>
+      <PopUpDelete title={card.title} />
```
Comment (sergeysova) [inline src/features/cards/organisms/card-item.js]: Maybe pass whole `card`?

```jsx
<CardDeletePopupButton card={card} />
```

_(just example)_

File: src/features/cards/organisms/card-item.js [hunk 4]
```diff
@@ -29,29 +83,115 @@ CardItem.propTypes = {
...
+      <Icon name="dots-v" height="1.6rem" />
     </Row>
   </HeadingLine>
 )
+const CardInfo = (card) => (
+  <Row>
+    <Text small>
+      {" "}
```
Comment (sergeysova) [inline src/features/cards/organisms/card-item.js]: Why you need `{" "}` here?

Can you add "space" with css/layout?

File: src/features/cards/organisms/card-item.js [hunk 5]
```diff
@@ -29,29 +83,115 @@ CardItem.propTypes = {
...
+const HeadingLine = styled.div`
+  display: flex;
+  flex-flow: row nowrap;
+  justify-content: space-between;
+  flex-shrink: 0;
+  line-height: 2.4rem;
+`
+const IconWrapper = styled.div`
+  padding: 0 0.6rem;
+  }
```
Comment (sergeysova) [inline src/features/cards/organisms/card-item.js]: Extra `}` ?

File: src/features/cards/organisms/card-item.js [hunk 6]
```diff
@@ -29,29 +83,115 @@ CardItem.propTypes = {
...
+  &:disabled {
+    opacity: 0.3;
+  }
+  
+  & #icon:hover {
```
Comment (sergeysova) [inline src/features/cards/organisms/card-item.js]: Please, don't use id selectors.

File: src/features/cards/organisms/card-item.js [hunk 7]
```diff
@@ -61,18 +201,107 @@ CardHeading.propTypes = {
...
 }
 
-const HeadingLine = styled.div`
-  display: flex;
-  flex-flow: row nowrap;
-  justify-content: space-between;
-  flex-shrink: 0;
-  line-height: 2.4rem;
-`
+const PopUpDelete = ({ title }) => {
```
Comment (sergeysova) [inline src/features/cards/organisms/card-item.js]: Please, reuse that code from #45

File: src/ui/atoms/icon.js [hunk 8]
```diff
@@ -0,0 +1,80 @@
+import React from "react"
+import PropTypes from "prop-types"
+// When you need more icons just add svg-path in getIconPath and call it
+// with <Icon name="yourname" />
+
+const getIconPath = (name, props) => {
```
Comment (sergeysova) [inline src/ui/atoms/icon.js]: I think we can use [`svg-react-loader`](https://github.com/jhamlet/svg-react-loader)
to load real .svg icons to react

Comment (meowsun) [reply src/ui/atoms/icon.js]: I am not satisfied with a way Icon component works now, especially hovers. I am going to fix/change work with svg in another branch/pr.

File: src/features/cards/organisms/card-item.js (hunk 6 above)
Comment (meowsun) [reply src/features/cards/organisms/card-item.js]: I am going to fix/change work with svg in another branch/pr.

Comment (sergeysova) [reply src/features/cards/organisms/card-item.js]: Maybe fix in that?

**Comment by sergeysova** [comment]
@meowsun Thank you!

**Comment by sergeysova** [comment]
@all-contributors please add @meowsun for code

**Pull request merged**